src/Projections/ProjectionStatistics.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * This file is part of prooph/event-store.
+ * (c) 2014-2020 Alexander Miertsch <kontakt@codeliner.ws>
+ * (c) 2015-2020 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Projections;
+
+use Webmozart\Assert\Assert;
+
+final class ProjectionStatistics
+{
+    private array $projections;
+
+    /**
+     * @param ProjectionDetails[]
+     */
+    public function __construct(array $projections)
+    {
+        Assert::allIsInstanceOf($projections, ProjectionDetails::class);
```

Comment:
not need, psalm will do static analysis

Reply:
This is not for static analysis, but for params validation, since array elements don't get runtime validation based on typehint. I understand that this is not necessary?

Reply:
Yes, we don't need this as we rely on psalm static analysis. It won't be checked on runtime.